mongodump/mongodump_test.go

```diff
@@ -1166,6 +1179,20 @@ func TestMongoDumpViewsAsCollections(t *testing.T) {
 
 			})
 
+			Convey("testing dumping a view query hints", func() {
```

Comment:
Two questions:
- Isn't it only supposed to hint the index when the storage engine is mmapv1?
- Should we also be testing this same assertion in `TestMongoDumpViews`?